### ISSUE TITLE
Copy clean links

### ISSUE THREAD

**Issue by oliversturm** [SimonBrazell/privacy-redirect#43]
Unless I'm missing something, this is a feature request. I'd like to be able to copy a link to the clipboard (for sharing elsewhere, usually) and have redirection rules applied to it. So I find a YouTube link in DuckDuckGo, right click and copy, end up with an invidious link in the clipboard.

Thoughts?

**Comment by SimonBrazell** [comment]
Hi @oliversturm I think this is a good idea, I'd have to look and see what sort of permissions it needs and how they are granted though as I want to avoid adding more to the extension if I can.

**Comment by oliversturm** [comment]
I believe clipboard-read and clipboard-write are separate permissions. Personally I wouldn't mind seeing an extension require clipboard-write alone, since that doesn't seem to be a privacy concern. But perhaps others would disagree.

**Comment by SimonBrazell** [comment]
Yeah and if it was something that was only allowed with a prompt then I think it would be fine, not sure if that's possible though, need to look into it.

**Comment by SimonBrazell** [comment]
I just know people will get annoyed if I start adding more permissions to the extension... which is why I'm a little wary.

**Comment by oliversturm** [comment]
Understood. Would be a pity if that problem was in the way of a cool feature. Idea: perhaps you can somehow show the modified link somewhere when I right-click, in a way that I can copy it myself. Then the extension wouldn't need extra permissions.

**Comment by SimonBrazell** [comment]
I'm sure there is a way to do it that will satisfy both camps, when I get some time I'll look into it, bit snowed under with other work lately...

**Comment by grravity** [comment]
It would be able to set a custom copy url, so if you run your own instance, and want to share a video through Invidious, you can set it to copy a public instance instead of your own.

Secondly, not sure if this would be out of the scope of this request, i but it would be cool to be able to copy the original url as well, so if I want to copy Invidious or the YouTube link, I can choose either quickly. The only way to copy the original YouTube link would be to open the Invidious page, and right click copy link on the Watch on YouTube button, or manually edit the URL. Does get kind of tedious after a while.

Just some ideas :)

**Comment by DJCrashdummy** [comment]
just to make it clear: @oliversturm with **clean** links you just mean the **rewritten** links?
...because for cleaning links from tracking tags there are other awesome addons like [ClearURLs](https://gitlab.com/KevinRoebert/ClearUrls) (which addresses even more URLs than just youtube, twitter, instagram & goggle maps).

if so, this may be an interesting (optional) feature... although i'm not sure about the UX.
if you are talking about also cleared URLs, i'm strongly against it because cramping everything into one addon is not a good idea! - better one addon does one job really good, than at ton of jobs but a bunch of them buggy.

**Comment by oliversturm** [comment]
> with **clean** links you just mean the **rewritten** links?

Yes, I meant "clean" in the sense of the rewriting done by this extension - not in the many senses used by other extensions.

I use ClearURLs myself and I like it, but I agree that there's no need to combine too many aspects of "cleaning" in one extension.

**Comment by SimonBrazell** [comment]
Okay looked into this a little further this evening and it appears to be fairly simple, it will need more permissions to access the context menus on right click and clipboard but these can be specified as optional so I'll be add an option to the extension options menu that enables this and prompts the user for the required permissions at the same time.

**Comment by smaragdus** [comment]
I use several copy/paste extensions and I don't mind clipboard permissions especially when the extensions are open source.

> Okay looked into this a little further this evening and it appears to be fairly simple, it will need more permissions to access the context menus on right click and clipboard but these can be specified as optional so I'll be able to add an option to the extension options menu that enables this and prompts the user for the required permissions at the same time.

Additional option for context menu would be great. Will copying apply to [Invidious](https://invidio.us/) links only or to all supported services?

**Comment by DJCrashdummy** [comment]
> Will copying apply to Invidious links only or to all supported services?

**if** such an (optional!) feature gets implemented, it should for sure work for all activated redirects.

**Comment by smaragdus** [comment]
@DJCrashdummy 

Thank you for answering my question. For me if implemented this option would be very handy and useful for fast copying of links without opening them.
